src/Microsoft.AspNetCore.Identity/UserManager.cs

```diff
@@ -1954,6 +1969,93 @@ public virtual Task<IList<TUser>> GetUsersInRoleAsync(string roleName)
         }
 
         /// <summary>
+        /// Returns an authentication token for a user.
+        /// </summary>
+        /// <param name="user"></param>
+        /// <param name="loginProvider">The authentication scheme for the provider the token is associated with.</param>
+        /// <param name="tokenName">The name of the token.</param>
+        /// <returns></returns>
+        public virtual Task<string> GetAuthenticationTokenAsync(TUser user, string loginProvider, string tokenName)
+        {
+            ThrowIfDisposed();
+            var store = GetTokenStore();
+            if (user == null)
+            {
+                throw new ArgumentNullException(nameof(user));
+            }
+            if (loginProvider == null)
+            {
+                throw new ArgumentNullException(nameof(loginProvider));
+            }
+            if (tokenName == null)
+            {
+                throw new ArgumentNullException(nameof(tokenName));
+            }
+
+            return store.GetTokenAsync(user, loginProvider, tokenName, CancellationToken);
+        }
+
+        /// <summary>
+        /// Sets an authentication token for a user.
+        /// </summary>
+        /// <param name="user"></param>
+        /// <param name="loginProvider">The authentication scheme for the provider the token is associated with.</param>
+        /// <param name="tokenName">The name of the token.</param>
+        /// <param name="tokenName">The value of the token.</param>
+        /// <returns></returns>
+        public virtual async Task<IdentityResult> SetAuthenticationTokenAsync(TUser user, string loginProvider, string tokenName, string tokenValue)
+        {
+            ThrowIfDisposed();
+            var store = GetTokenStore();
+            if (user == null)
+            {
+                throw new ArgumentNullException(nameof(user));
+            }
+            if (loginProvider == null)
+            {
+                throw new ArgumentNullException(nameof(loginProvider));
+            }
+            if (tokenName == null)
+            {
+                throw new ArgumentNullException(nameof(tokenName));
+            }
+
+            // REVIEW: should updating any tokens affect the security stamp?
```

Comment:
Don't think so If we get a new facebook token that's not a reason to invalidate a cookie login elsewhere is it?